src/test/net/DatabaseDownloader_test.cpp

```diff
@@ -29,6 +28,8 @@
 namespace ripple {
 namespace test {
 
+#define REPORT_FAILURE(D) reportFailure(D, __FILE__, __LINE__);
```

Comment:
After a night's sleep, I have a couple of suggestions regarding the `REPORT_FAILURE` macro.

A) Macros work by text substitution.  Since you're including the semicolon in your macro, the places that invoke your macro and include the terminating semicolon actually have two semicolons — the explicit semicolon plus the one from the macro.  That's not the end of the world.

However this also allows the person who's maintaining the code to accidentally omit the explicit closing semicolon.  That's also not the end of the world, but it looks weird to other people reading the code.

You can avoid all these concerns by removing the closing semicolon from the `REPORT_FAILURE` macro.  Consider it.

B) Macros have no scope.  So once they are defined they affect everything that follows in the transaction unit.  We usually expect a translation unit to end at the conclusion a `*.cpp` file.  So we don't expect a macro that's defined below the `#include`s in a `*.cpp` file to leak into any other files.

However rippled can optionally be built in a unity style.  For unity builds multiple `*.cpp` files are concatenated into a single translation unit.  Suddenly the macro you thought was only affecting your own `*.cpp` file has leaked into other `*.cpp` files.  That _usually_ doesn't create any problems.  But when it does those kinds of problems can be really hard to figure out.

The way to avoid complications like this in unity builds is to explicitly "scope" your macro.  I suggest you `#undef REPORT_FAILURE` near the end of the file.  This way the macro will not be able to leak into other files.

**Suggestions Summary:**
1. Remove the closing semicolon from the `REPORT_FAILURE` macro.
2. `#undef REPORT_FAILURE` toward the bottom of the file.

Sorry I didn't catch these earlier.

Reply:
Good catch

Reply:
Done